homeassistant/helpers/translation.py

```diff
@@ -165,20 +165,21 @@ async def _async_get_component_strings(
     for language in languages:
         files_to_load: dict[str, str] = {}
         files_to_load_by_language[language] = files_to_load
-
-        loaded_translations: dict[str, Any] = {}
-        translations_by_language[language] = loaded_translations
+        translations_by_language[language] = {}
 
         for loaded in components:
-            domain = loaded.partition(".")[0]
+            domain, _, platform = loaded.partition(".")
             if not (integration := integrations.get(domain)):
                 continue
 
-            path = component_translation_path(loaded, language, integration)
-            # No translation available
-            if path is None:
-                loaded_translations[loaded] = {}
-            else:
+            if platform and integration.is_built_in:
+                # Legacy state translations are no longer used for built-in integrations
+                # and we avoid trying to load them. This is a temporary measure to allow
+                # them to keep working for custom integrations until we can fully remove
+                # them.
+                continue
+
+            if path := component_translation_path(loaded, language, integration):
                 files_to_load[loaded] = path
 
     if not files_to_load:
```

Comment:
This check only looks at the last language that was processed. It should only abort if none of the languages have anything to load.

```suggestion
    if not any(paths for paths in translations_by_language.items()):
```

Reply:
Should this suggestion be
`if not any(paths for paths in files_to_load_by_language.values())`?

Reply:
I think I should do a separate bugfix pr fix the scope bug with a test as that affects production.  I'm not sure it is an actual problem or not yet.

Reply:
I don't think it ever happens, English is always passed in first and is loaded.

Reply:
I think the only way it could happen is hitting this path. Maybe not event then

If it's a component that is just one file, we don't support translations
    # Example custom_components/my_component.py

Reply:
I guess if we try to load translations for an integration that is missing but it's going to be missing for both English and the other locale

Reply:
> I think I should do a separate bugfix pr fix the scope bug with a test as that affects production. I'm not sure it is an actual problem or not yet.

The check inside the loop will always be the same regardless of the language since the only checks are for an the integration/manifest and that cannot change between languages. Even though it works, its still a scope bug so https://github.com/home-assistant/core/pull/112457